src/main/java/org/kuali/kfs/sys/rest/resource/businessobject/BusinessObjectsControllersHelperService.java

```diff
@@ -0,0 +1,423 @@
+    ) {
+        LOG.debug("getLookupValues(...) - Enter : businessObjectName={}", businessObjectName);
+
+        final BusinessObjectEntry businessObjectEntry =
+                businessObjectDictionaryService.getBusinessObjectEntry(businessObjectName);
+
+        ensureAuthorization(userSession, businessObjectEntry);
+
+        final Map<String, Object> valuesMap = new LinkedHashMap<>();
+
+        final Class<? extends BusinessObjectBase> businessObjectBaseClass =
+                (Class<? extends BusinessObjectBase>) businessObjectEntry.getBusinessObjectClass();
+        final SearchService searchService = getSearchService(businessObjectBaseClass);
+        final List<FormAttribute> attributes = searchService.getFormAttributes(businessObjectBaseClass);
+        for (final FormAttribute attribute: attributes) {
+            final Control control = attribute.getControl();
+            if (control == null) {
+                continue;
+            }
+
+            final String singleAttributeName = attribute.getName();
+
+            if (control.getType() == Control.Type.TREE) {
+                // we have to do this bean resolution here b/c batch file (the only tree) is still a snowflake
+                // and the DDD doesn't do the bean lookup for us (mainly b/c of the typing); hope to get rid of the
+                // need for special VF type eventually
+                final String valuesFinderName = control.getValuesFinderName();
+                if (StringUtils.isBlank(valuesFinderName)) {
+                    LOG.warn(
+                            "getLookupValues(...) - Cannot find a ValuesFinder : businessObjectName={}; attributename={}",
+                            businessObjectEntry::getName,
+                            () -> singleAttributeName
+                    );
+                    continue;
+                }
+                final HierarchicalControlValuesFinder valuesFinder =
+                        dataDictionaryService.getDDBean(HierarchicalControlValuesFinder.class, valuesFinderName);
+                if (valuesFinder == null) {
+                    LOG.warn(
+                            "getLookupValues(...) - Cannot find a HierarchicalControlValuesFinder : businessObjectName={}; attributename={}",
+                            businessObjectEntry::getName,
+                            () -> singleAttributeName
+                    );
+                    continue;
+                }
+                final List<HierarchicalData> values = valuesFinder.getHierarchicalControlValues();
+                valuesMap.put(singleAttributeName, values);
+            } else {
+                final KeyValuesFinder valuesFinder = control.getValuesFinder();
+                if (valuesFinder == null) {
+                    continue;
+                }
+                // CU Customization: keyValues list now comes from the helper method below.
+                final List<KeyValue> keyValues = getKeyValuesForLookup(valuesFinder);
```

Comment:
would it make more sense if this block was included in the if statement above and the log was changed to valuesFinder != null

Reply:
In my opinion, such restructuring would result in unnecessary deviation from how KualiCo had coded this base method. The subsequent put-in-map line needs to be skipped if the values finder is null, meaning there's more than just this one customized line that needs adjustments if we modify the if-null-then-continue block above. Or are you referring to some other type of refactoring here?

Reply:
ah, I think I misunderstood somthing, lets leave it as you coded it